File: frontend/lib/utils/logger.dart
import 'package:flutter/foundation.dart' show debugPrint, kDebugMode;

class Logger {
  static void debug(String message, [dynamic error]) {
    if (kDebugMode) {
      debugPrint('[DEBUG] $message${error != null ? ' - $error' : ''}');
    }
  }

  static void info(String message, [dynamic error]) {
    if (kDebugMode) {
      debugPrint('[INFO] $message${error != null ? ' - $error' : ''}');
    }
  }

  static void warning(String message, [dynamic error]) {
    if (kDebugMode) {
      debugPrint('[WARNING] $message${error != null ? ' - $error' : ''}');
    }
  }

  static void error(String message, [dynamic error]) {
    if (kDebugMode) {
      debugPrint('[ERROR] $message${error != null ? ' - $error' : ''}');
      if (error != null && error is Error) {
        debugPrint(error.stackTrace.toString());
      }
    }
  }
}

File: frontend/src/shared/utils/logger.ts
const isDev = __DEV__;

const fmt = (level: string, message: string, error?: unknown) =>
  `[${level}] ${message}${error !== undefined ? ` - ${String(error)}` : ''}`;

export const Logger = {
  debug(message: string, error?: unknown) {
    if (isDev) console.log(fmt('DEBUG', message, error));
  },
  info(message: string, error?: unknown) {
    if (isDev) console.log(fmt('INFO', message, error));
  },
  warning(message: string, error?: unknown) {
    if (isDev) console.warn(fmt('WARNING', message, error));
  },
  error(message: string, error?: unknown) {
    if (isDev) console.error(fmt('ERROR', message, error));
  },
};
